File: test/cpp/polyglot/times/portions.cpp
// Copyright 2015 Adam Grandquist

#include "./catch.hpp"
#include "./test.hpp"
#include "./ReQL.hpp"

using namespace ReQL;
using namespace _C;

TEST_CASE("cpp accessing portions", "[cpp][ast]") {

  SECTION("test10") {
    Types::array arr0(4);

    double num1(1375142400);
    Query var1(num1);

    arr0.insert(arr0.end(), var1);

    double num2(1375142400);
    Query var2(num2);

    arr0.insert(arr0.end(), var2);

    double num3(1375142400);
    Query var3(num3);

    arr0.insert(arr0.end(), var3);

    double num4(2375136000);
    Query var4(num4);

    arr0.insert(arr0.end(), var4);

    Query var0(arr0);
  }

  SECTION("test11") {
    Types::array arr0(4);

    double num1(0);
    Query var1(num1);

    arr0.insert(arr0.end(), var1);

    double num2(0);
    Query var2(num2);

    arr0.insert(arr0.end(), var2);

    double num3(0);
    Query var3(num3);

    arr0.insert(arr0.end(), var3);

    double num4(0);
    Query var4(num4);

    arr0.insert(arr0.end(), var4);

    Query var0(arr0);
  }

  SECTION("test12") {
    Types::array arr0(4);

    double num1(4896.681);
    Query var1(num1);

    arr0.insert(arr0.end(), var1);

    double num2(4896.682);
    Query var2(num2);

    arr0.insert(arr0.end(), var2);

    double num3(4897.681);
    Query var3(num3);

    arr0.insert(arr0.end(), var3);

    double num4(11296.681);
    Query var4(num4);

    arr0.insert(arr0.end(), var4);

    Query var0(arr0);
  }

  SECTION("test13") {
    Types::array arr0(4);

    Types::array arr1(6);

    double num2(2013);
    Query var2(num2);

    arr1.insert(arr1.end(), var2);

    double num3(7);
    Query var3(num3);

    arr1.insert(arr1.end(), var3);

    double num4(30);
    Query var4(num4);

    arr1.insert(arr1.end(), var4);

    double num5(1);
    Query var5(num5);

    arr1.insert(arr1.end(), var5);

    double num6(21);
    Query var6(num6);

    arr1.insert(arr1.end(), var6);

    double num7(36.681);
    Query var7(num7);

    arr1.insert(arr1.end(), var7);

    Query var1(arr1);

    arr0.insert(arr0.end(), var1);

    Types::array arr8(6);

    double num9(2013);
    Query var9(num9);

    arr8.insert(arr8.end(), var9);

    double num10(7);
    Query var10(num10);

    arr8.insert(arr8.end(), var10);

    double num11(30);
    Query var11(num11);

    arr8.insert(arr8.end(), var11);

    double num12(1);
    Query var12(num12);

    arr8.insert(arr8.end(), var12);

    double num13(21);
    Query var13(num13);

    arr8.insert(arr8.end(), var13);

    double num14(36.682);
    Query var14(num14);

    arr8.insert(arr8.end(), var14);

    Query var8(arr8);

    arr0.insert(arr0.end(), var8);

    Types::array arr15(6);

    double num16(2013);
    Query var16(num16);

    arr15.insert(arr15.end(), var16);

    double num17(7);
    Query var17(num17);

    arr15.insert(arr15.end(), var17);

    double num18(30);
    Query var18(num18);

    arr15.insert(arr15.end(), var18);

    double num19(1);
    Query var19(num19);

    arr15.insert(arr15.end(), var19);

    double num20(21);
    Query var20(num20);

    arr15.insert(arr15.end(), var20);

    double num21(37.681);
    Query var21(num21);

    arr15.insert(arr15.end(), var21);

    Query var15(arr15);

    arr0.insert(arr0.end(), var15);

    Types::array arr22(6);

    double num23(2045);
    Query var23(num23);

    arr22.insert(arr22.end(), var23);

    double num24(4);
    Query var24(num24);

    arr22.insert(arr22.end(), var24);

    double num25(7);
    Query var25(num25);

    arr22.insert(arr22.end(), var25);

    double num26(3);
    Query var26(num26);

    arr22.insert(arr22.end(), var26);

    double num27(8);
    Query var27(num27);

    arr22.insert(arr22.end(), var27);

    double num28(16.681);
    Query var28(num28);

    arr22.insert(arr22.end(), var28);

    Query var22(arr22);

    arr0.insert(arr0.end(), var22);

    Query var0(arr0);
  }

  SECTION("test14") {
    Types::object map0;

    std::string src1("func", 4);
    Query var1(src1);

    std::string src2("rts", 3);
    Query var2(src2);

    map0.insert({src1, var2});

    Query var0(map0);
  }

  SECTION("test15") {
    double num0(0);
    Query var0(num0);
  }

  SECTION("test16") {
    Types::array arr0(4);

    Types::array arr1(2);

    double num2(2);
    Query var2(num2);

    arr1.insert(arr1.end(), var2);

    double num3(211);
    Query var3(num3);

    arr1.insert(arr1.end(), var3);

    Query var1(arr1);

    arr0.insert(arr0.end(), var1);

    Types::array arr4(2);

    double num5(2);
    Query var5(num5);

    arr4.insert(arr4.end(), var5);

    double num6(211);
    Query var6(num6);

    arr4.insert(arr4.end(), var6);

    Query var4(arr4);

    arr0.insert(arr0.end(), var4);

    Types::array arr7(2);

    double num8(2);
    Query var8(num8);

    arr7.insert(arr7.end(), var8);

    double num9(211);
    Query var9(num9);

    arr7.insert(arr7.end(), var9);

    Query var7(arr7);

    arr0.insert(arr0.end(), var7);

    Types::array arr10(2);

    double num11(5);
    Query var11(num11);

    arr10.insert(arr10.end(), var11);

    double num12(97);
    Query var12(num12);

    arr10.insert(arr10.end(), var12);

    Query var10(arr10);

    arr0.insert(arr0.end(), var10);

    Query var0(arr0);
  }
}

File: libReQLTests/polyglot/times/portions.m
// Copyright 2015 Adam Grandquist

#import <Cocoa/Cocoa.h>
#import <XCTest/XCTest.h>
#import <libReQL.h>

@interface ObjcAccessingPortions : XCTestCase

@end

@implementation ObjcAccessingPortions


- (void)test10 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:4];
  NSNumber *var1 = @1375142400;
  [var0 addObject:var1];
  NSNumber *var2 = @1375142400;
  [var0 addObject:var2];
  NSNumber *var3 = @1375142400;
  [var0 addObject:var3];
  NSNumber *var4 = @2375136000;
  [var0 addObject:var4];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test11 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:4];
  NSNumber *var1 = @0;
  [var0 addObject:var1];
  NSNumber *var2 = @0;
  [var0 addObject:var2];
  NSNumber *var3 = @0;
  [var0 addObject:var3];
  NSNumber *var4 = @0;
  [var0 addObject:var4];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test12 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:4];
  NSNumber *var1 = @4896.681;
  [var0 addObject:var1];
  NSNumber *var2 = @4896.682;
  [var0 addObject:var2];
  NSNumber *var3 = @4897.681;
  [var0 addObject:var3];
  NSNumber *var4 = @11296.681;
  [var0 addObject:var4];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test13 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:4];
  NSMutableArray *var1 = [NSMutableArray arrayWithCapacity:6];
  NSNumber *var2 = @2013;
  [var1 addObject:var2];
  NSNumber *var3 = @7;
  [var1 addObject:var3];
  NSNumber *var4 = @30;
  [var1 addObject:var4];
  NSNumber *var5 = @1;
  [var1 addObject:var5];
  NSNumber *var6 = @21;
  [var1 addObject:var6];
  NSNumber *var7 = @36.681;
  [var1 addObject:var7];
  [var0 addObject:var1];
  NSMutableArray *var8 = [NSMutableArray arrayWithCapacity:6];
  NSNumber *var9 = @2013;
  [var8 addObject:var9];
  NSNumber *var10 = @7;
  [var8 addObject:var10];
  NSNumber *var11 = @30;
  [var8 addObject:var11];
  NSNumber *var12 = @1;
  [var8 addObject:var12];
  NSNumber *var13 = @21;
  [var8 addObject:var13];
  NSNumber *var14 = @36.682;
  [var8 addObject:var14];
  [var0 addObject:var8];
  NSMutableArray *var15 = [NSMutableArray arrayWithCapacity:6];
  NSNumber *var16 = @2013;
  [var15 addObject:var16];
  NSNumber *var17 = @7;
  [var15 addObject:var17];
  NSNumber *var18 = @30;
  [var15 addObject:var18];
  NSNumber *var19 = @1;
  [var15 addObject:var19];
  NSNumber *var20 = @21;
  [var15 addObject:var20];
  NSNumber *var21 = @37.681;
  [var15 addObject:var21];
  [var0 addObject:var15];
  NSMutableArray *var22 = [NSMutableArray arrayWithCapacity:6];
  NSNumber *var23 = @2045;
  [var22 addObject:var23];
  NSNumber *var24 = @4;
  [var22 addObject:var24];
  NSNumber *var25 = @7;
  [var22 addObject:var25];
  NSNumber *var26 = @3;
  [var22 addObject:var26];
  NSNumber *var27 = @8;
  [var22 addObject:var27];
  NSNumber *var28 = @16.681;
  [var22 addObject:var28];
  [var0 addObject:var22];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test14 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableDictionary *var0 = [NSMutableDictionary dictionaryWithCapacity:1];
  NSString *var1 = @"func";
  NSString *var2 = @"rts";
  [var0 setObject:var2 forKey:var1];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test15 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSNumber *var0 = @0;
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test16 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:4];
  NSMutableArray *var1 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var2 = @2;
  [var1 addObject:var2];
  NSNumber *var3 = @211;
  [var1 addObject:var3];
  [var0 addObject:var1];
  NSMutableArray *var4 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var5 = @2;
  [var4 addObject:var5];
  NSNumber *var6 = @211;
  [var4 addObject:var6];
  [var0 addObject:var4];
  NSMutableArray *var7 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var8 = @2;
  [var7 addObject:var8];
  NSNumber *var9 = @211;
  [var7 addObject:var9];
  [var0 addObject:var7];
  NSMutableArray *var10 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var11 = @5;
  [var10 addObject:var11];
  NSNumber *var12 = @97;
  [var10 addObject:var12];
  [var0 addObject:var10];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

@end
